### PR TITLE
feat: show cluster endpoints with hub status

### PR DESCRIPTION
**Pull request by sshaman1101**
added:
- hub: Hide the announcer behind the abstraction.
- hub: the announcements can be fired when timer ticks, when .Once() is called by hand.
- proto: extend hub status message with cluster endpoints.
- cli: print Hub's endpoints and announcement status.

### CODE REVIEW

**Comment by sshaman1101** [comment]
@oopcode @3Hren @antmat PTAL!

File: insonmnia/hub/announcer.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	return &locatorAnnouncer{
+		key:     key,
+		cluster: cls,
+		client:  lc,
+		period:  td,
+	}
+}
+
+func (la *locatorAnnouncer) Start(ctx context.Context) {
+	tk := time.NewTicker(la.period)
```
Comment (antmat) [inline insonmnia/hub/announcer.go]: util.ImmediateTicker?

File: insonmnia/hub/announcer.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	log "github.com/noxiouz/zapctx/ctxlog"
+	pb "github.com/sonm-io/core/proto"
+	"github.com/sonm-io/core/util"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/status"
+)
+
+type Announcer interface {
+	// Start starts announcement
+	Start(ctx context.Context)
```
Comment (antmat) [inline insonmnia/hub/announcer.go]: Maybe AnnounceLoop and Announce would be better?

File: insonmnia/hub/server.go [hunk 3]
```diff
@@ -1073,7 +1085,10 @@ func (h *Hub) Serve() error {
 
 	h.waiter.Go(h.runCluster)
 	h.waiter.Go(h.listenClusterEvents)
-	h.waiter.Go(h.startLocatorAnnouncer)
+	h.waiter.Go(func() error {
```
Comment (antmat) [inline insonmnia/hub/server.go]: I'd personally leave startLocatorAnnouncer wrapper, cause waiter block would look more compact

File: insonmnia/hub/server.go [hunk 4]
```diff
@@ -1179,57 +1194,19 @@ func (h *Hub) GetMinerByID(minerID string) *MinerCtx {
...
-func (h *Hub) announceAddress() error {
-	if !h.cluster.IsLeader() {
-		return nil
-	}
-
-	members, err := h.cluster.Members()
+	members, err := cluster.Members()
 	if err != nil {
-		return err
+		return clients, workers, err
```
Comment (antmat) [inline insonmnia/hub/server.go]: Y U NO explicitly nil, nil, err?

File: insonmnia/hub/announcer.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	defer la.mu.Unlock()
+
+	la.err = err
+}
+
+func (la *locatorAnnouncer) ErrorMsg() string {
+	la.mu.Lock()
+	defer la.mu.Unlock()
+
+	s, ok := status.FromError(la.err)
```
Comment (antmat) [inline insonmnia/hub/announcer.go]: maybe just check for nil?

File: proto/hub.proto [hunk 6]
```diff
@@ -106,6 +106,10 @@ message HubStatusReply {
     string version = 3;
     string platform = 4;
     string ethAddr = 5;
+
+    string clientEndpoint = 6;
```
Comment (antmat) [inline proto/hub.proto]: btw, why we have only one client endpoint?

**Pull request merged**